src/questions/data_types.yaml

```
@@ -1,37 +1,1106 @@
+  explanation: x << y is equivalent to the mathematical expression x * 2**y.
+  answerKeys:
+    - D
+  sub_topics:
+    - "value-types"
+  difficulty_level: Hard
+  choices:
+    - content: "0"
+      key: A
+    - content: "1"
+      key: B
+    - content: "8"
+      key: C
+    - content: "12"
+      key: D
+
+- uuid: 203d97e1-7194-4373-b943-d89d06e35126
+  type: MultipleChoice
+  content: Which of the following are valid literals for solidity data types?
+  hint: noHint
+  explanation: 2.5e1 is an integer literal , "foo" "bar" is equivalent to “foobar” which is a string literal and hex"001122FF" is an example of  hexadecimal literal which are prefixed with the keyword hex.
+  answerKeys:
+    - A
+    - B
+    - C
+  sub_topics:
+    - "value-types"
+  difficulty_level: Low
+  choices:
+    - content: 2.5e1
+      key: A
+    - content: " 'foo' 'bar' "
+      key: B
+    - content: hex"001122FF"
+      key: C
+    - content: None of these
+      key: D
+
+- uuid: 2bb12876-aee8-4cbd-8662-e193bb5503aa
+  type: SingleChoice
+  content: "Variable packing is one way solidity optimizes gas costs in storage state variables . Consider the following two examples of variable declarations.
+            Example 1 -  uint128 a;
+                         uint256 b;
+                         uint128 c;
+            Example 2 -  uint128 a;
+                         uint128 c;
+                         uint256 b;
+            Which of the following statements are correct regarding variable packing in the above examples."
+  hint: Solidity contracts have contiguous 32 byte (256 bit) slots used for storage.
+  explanation: In example 1 the variables are not packed. If b was packed with a, it would exceed the 32 byte limit so it is instead placed in a new storage slot. The same thing happens with c and b. In example 2 the  variables are packed. Because packing c with a does not exceed the 32 byte limit, they are stored in the same slot.
+  answerKeys:
+    - C
+  sub_topics:
+    - "value-types"
+  difficulty_level: Hard
+  choices:
+    - content: Variables in both examples 1 and 2 are packed.
+      key: A
+    - content: Variables in example 1 are packed but not in example 2.
+      key: B
+    - content: Variables in example 2 are packed but not in example 1.
+      key: C
+    - content: Variables in neither examples 1 and 2 are packed.
+      key: D
+
+- uuid: 47e3f7e4-b2e6-4c0f-acd7-e7c960203710
+  type: SingleChoice
+  content: "Consider the following code snippet ,
+              // SPDX-License-Identifier: GPL-3.0
+                pragma solidity ^0.8.8;
+                contract test {
+                  enum ActionChoices { GoLeft, GoRight, GoStraight, SitStill }
+
+                  function foo() public pure returns (ActionChoices) {
+                      return type(ActionChoices).max;
+                  }
+                }
+            What will be the output when the function foo is called ?"
```

````suggestion
  content: | 
         Consider the following code snippet ,
         ```
              // SPDX-License-Identifier: GPL-3.0
                pragma solidity ^0.8.8;
                contract test {
                  enum ActionChoices { GoLeft, GoRight, GoStraight, SitStill }

                  function foo() public pure returns (ActionChoices) {
                      return type(ActionChoices).max;
                  }
                }
            What will be the output when the function foo is called ?
         ```   
````